madprops/serializers.py

```diff
@@ -61,95 +89,123 @@ class Property(model):
         parent_obj_field = self.opts.parent_obj_field
         filters = {
-            parent_obj_field: getattr(obj, parent_obj_field),
-            'name': obj.name
+            parent_obj_field: property_data[parent_obj_field],
+            'name': prop_name
         }
-        if not model.objects.filter(**filters).update(value=obj.value):
-            obj.save(**kwargs)
 
+        # If it already exists - update it's value. Otherwise - create a new
+        # property.
+        prop = self.Meta.model.objects.filter(**filters).first()
 
-class NestedPropertySerializer(PropertySerializer):
-    """Version of PropertySerializer for nested resources
+        if prop_name in self.opts.read_only_props:
+            return prop
 
-    Intended to be used as a base class for serializer for property resource
-    exposed as a nested resource.
-    """
+        if prop:
+            prop.value = property_data['value']
+            prop.save()
+        else:
+            prop = self.Meta.model.objects.create(**property_data)
+
+        return prop
+
+    def to_internal_value(self, data):
+        data = self._to_extended_dict(data)
 
-    _options_class = NestedPropertySerializerOptions
+        # Handle JSON fields (value encoded as JSON).
+        if data['name'] in self.opts.json_props:
+            data['value'] = json.dumps(data['value'])
 
-    def __init__(self, **kwargs):
-        super(NestedPropertySerializer, self).__init__(**kwargs)
-        self.many = True
+        data = self._add_parent_obj_field(data)
+        return super(PropertySerializer, self).to_internal_value(data)
 
-    def _from_native_hook(self, data):
+    def _to_extended_dict(self, data):
+        """ Convert dictionary of properties:
+        {<prop_name>: <prop_value>} ->
+            {'name': <prop_name>, 'value': <prop_value>}
+        """
+        prop_name, prop_value = data.items()[0]
+        return {'name': prop_name, 'value': prop_value}
+
+    def _add_parent_obj_field(self, data):
+        # Property requires ID of parent object (properties owner, e.g. User
+        # for user's properties case). Take it from context, which is passed
+        # from view.
+        parent_obj_field = self.opts.parent_obj_field
+        parent_id_field = parent_obj_field + '_id'
+        data[parent_obj_field] = self.context.get(
+            'parent_id') or self.context['view'].kwargs.get(
+                parent_id_field)
         return data
+
+
+class PropertiesOwnerSerializer(ModelSerializer):
+    """Base class for "parent" serializers.
+
+    If you want to use PropertySerializer with a parent model (e.g. as a
+    nested serializer for User serializer), and have this field writable  -
+    you have to inherit you parent serializer from this class, not
+    ModelSerializer.
+    """
+    def create(self, validated_data):
+        validated_data_minus_properties = dict(validated_data)
+        properties_field = None
+        for (field, serializer) in validated_data:
```

Comment:
missed `items()` also, shouldn't we iterate over `self.fields.items()` to find which field is our properties field?
I thought `validated_data` have dict of properties as value not serializer.